battlebots/database/access.py

```diff
@@ -37,7 +37,8 @@ def remove_bot(user, botname):
                         % (user.nickname, botname, code_dir))
         pass
 
-    bot = session.query(Bot).filter_by(user=user, name=botname).one_or_none()
```

Comment:
The commit/rollback part would indeed be unnecessary, so maybe we can make two kinds of context managers? Or maybe change `Session.query(...)` to `Session().query(...)`?